reconcile/test/cna/test_integration.py

```diff
@@ -144,12 +153,23 @@ def null_asset(name: str, addr_block: Optional[str]) -> NullAsset:
     ],
 )
 def test_integration_assemble_current_states(
-    cna_clients: Mapping[str, CNAClient],
+    mocker,
     listed_assets: Iterable[Mapping[str, Any]],
     expected_state: State,
 ):
-    cna_clients["test"].list_assets.side_effect = [listed_assets]  # type: ignore
-    integration = CNAIntegration(cna_clients=cna_clients, namespaces=[])
+    mocker.patch.object(
+        CNAClient, "list_assets", create_autospec=True, return_value=listed_assets
```

Comment:
`create_autospec` is a method whereas `autospec` should be the option to enabling autospeccing for the object created by the patch. I'm actually a bit surprised that no errors are thrown by this unless I'm missing something?

I confirmed with `patch` that `create_autospec` doesn't do what we expected, while `autospec` does:

```python
>>> from unittest.mock import patch
>>> def test(a):
...     print(a)
... 

# In this test too many arguments are passed, which should result in an error, but it works just fine
>>> with patch('__main__.test', return_value="test", create_autospec=True) as patch_test:
...     print(test('too', 'many', 'args'))
... 
test

# With autospec=True we get the expected TypeError
>>> with patch('__main__.test', return_value="test", autospec=True) as patch_test:
...     print(test('too', 'many', 'args'))
... 
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
  File "<string>", line 2, in test
  File "/usr/lib64/python3.9/unittest/mock.py", line 180, in checksig
    sig.bind(*args, **kwargs)
  File "/usr/lib64/python3.9/inspect.py", line 3043, in bind
    return self._bind(args, kwargs)
  File "/usr/lib64/python3.9/inspect.py", line 2964, in _bind
    raise TypeError('too many positional arguments') from None
TypeError: too many positional arguments
```